support-modules/supporter-product-data-dynamo/src/main/scala/com/gu/supporterdata/model/SupporterRatePlanItem.scala

```diff
@@ -10,4 +10,7 @@ case class SupporterRatePlanItem(
     productRatePlanName: String, // Name of the product in this rate plan
     termEndDate: LocalDate, // Date that this subscription term ends
     contractEffectiveDate: LocalDate, // Date that this subscription started
+    contributionAmount: Option[ContributionAmount],
 )
+
+case class ContributionAmount(amount: BigDecimal, currency: String)
```

Comment:
There is also a `Price` model in membership-common, but it uses Float, which I think is more susceptible than BigDecimal? - https://github.com/guardian/membership-common/blob/main/src/main/scala/com/gu/memsub/Price.scala#L7

Reply:
oh interesting, having read into it, bigdecimal actually scales the number in base 10 at whatever precision is necessary (rather than base 2 with limited precision), and it would throw an exception if it couldn't represent the number exactly.
In our case, it basically does the trick I mentioned for free.

Reply:
and tom yes I think it was back in the membership-common days that we had trouble with discounting or something due to floats.